projects/gui/src/tournamentsettingswidget.cpp

```diff
@@ -32,6 +32,73 @@ TournamentSettingsWidget::TournamentSettingsWidget(QWidget *parent)
 		ui->m_seedsSpin->setEnabled(checked);
 	});
 
+	// number of rounds limits repeats
+	connect(ui->m_roundsSpin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
+		[=](int value)
+	{
+		if (!ui->m_gauntletRadio->isChecked())
+			value = 1;
+		int limit = value * ui->m_gamesPerEncounterSpin->value();
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// number of games per encounter limits repeats
+	connect(ui->m_gamesPerEncounterSpin,
+		static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), [=](int value)
+	{	int rounds = ui->m_gauntletRadio->isChecked() ?  ui->m_roundsSpin->value() : 1;
+		int limit = value * rounds;
+		limit = (limit > 999999 ) ? 999999 : limit;
+		ui->m_repeatSpin->setMaximum(limit);
+	});
+
+	// update repeats after rounds
+	connect(ui->m_roundsSpin,
+		static_cast<void (QSpinBox::*)()>(&QAbstractSpinBox::editingFinished), [=]()
+	{
+		int value = ui->m_repeatSpin->value();
+		ui->m_repeatSpin->valueChanged(value);
```

Comment:
It's okay to call slots directly but I think signals should always be used with the `emit signal(param);` syntax.

Reply:
Changed accordingly.